testinfra/modules/socket.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf8 -*-
+# Copyright © 2015 Philippe Pepiot
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import unicode_literals
+
+from testinfra.modules.base import Module
+
+
+def parse_socketspec(socketspec):
+
+    protocol, address = socketspec.split("://", 1)
+
+    if protocol not in ("udp", "tcp", "unix"):
+        raise NotImplementedError("Protocol %s" % (protocol,))
+
+    if protocol == "unix":
+        # unix:///foo/bar.sock
+        host = address
+        port = None
+    elif ":" in address:
+        # tcp://127.0.0.1:22
+        # tcp://:::22
+        host, port = address.rsplit(":", 1)
+        port = int(port)
+    else:
+        # tcp://22
+        host = None
+        port = int(address)
+
```

Comment:
Indeed. I've to check adress and port. I think this do the trick: https://stackoverflow.com/questions/319279/how-to-validate-ip-address-in-python/319298#319298